src/core/Alert/InlineAlert/InlineAlert.tsx

```diff
@@ -42,30 +55,34 @@ class BaseInlineAlert extends Component<InlineAlertProps & InnerRef> {
       <HtmlDivWithRef
         as="section"
         {...passProps}
-        className={classnames(
-          baseClassName,
-          alertClassNames.inline,
-          className,
-          {
-            [`${baseClassName}--${status}`]: !!status,
-            [alertClassNames.smallScreen]: !!smallScreen,
-          },
-        )}
+        className={classnames(baseClassName, className, {
+          [`${baseClassName}--${status}`]: !!status,
+          [inlineAlertClassNames.smallScreen]: !!smallScreen,
+        })}
       >
-        <HtmlDiv className={alertClassNames.styleWrapper}>
+        <HtmlDiv className={inlineAlertClassNames.styleWrapper}>
           {status !== 'neutral' && (
-            <Icon icon={status} className={alertClassNames.icon} />
+            <Icon
+              icon={status}
+              className={classnames(inlineAlertClassNames.icon, {
+                [`${inlineAlertClassNames.icon}--${status}`]: !!status,
```

Comment:
Adding status based classnames, e.g `fi-inline-alert--error` to be more specific and not to bleed the icon colouring to nested content.